### PR TITLE
in_http: changes for lambda extension logs support

### PR DESCRIPTION
**Pull request by PettitWesley**
Signed-off-by: Wesley Pettit <wppttt@amazon.com>

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:
- [ ] Example configuration file for the change
- [ ] Debug log output from testing the change

- [ ] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

**Documentation**

- [ ] Documentation required for this feature

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

### CODE REVIEW

File: plugins/in_http/http_prot.c [hunk 1]
```diff
@@ -76,37 +76,72 @@ int process_pack(struct flb_http *ctx, flb_sds_t tag, char *buf, size_t size)
...
+    msgpack_object record;
 
     flb_time_get(&tm);
 
     msgpack_unpacked_init(&result);
     while (msgpack_unpack_next(&result, buf, size, &off) == MSGPACK_UNPACK_SUCCESS) {
-        if (result.data.type != MSGPACK_OBJECT_MAP) {
-            flb_plg_warn(ctx->ins, "skip record from invalid type: %i",
+        if (result.data.type == MSGPACK_OBJECT_MAP) {
+            flb_info("map!");
```
Comment (edsiper) [inline plugins/in_http/http_prot.c:87]: :information_source:  ?

File: plugins/in_http/http_prot.c [hunk 2]
```diff
@@ -76,37 +76,72 @@ int process_pack(struct flb_http *ctx, flb_sds_t tag, char *buf, size_t size)
...
+            }
+            else {
+                /* use default plugin Tag (it internal name, e.g: http.0 */
+                flb_input_chunk_append_raw(ctx->ins, NULL, 0, mp_sbuf.data, mp_sbuf.size);
+            }
+            msgpack_sbuffer_destroy(&mp_sbuf);
+
+            break;
+    
+        } else if (result.data.type == MSGPACK_OBJECT_ARRAY) {
```
Comment (edsiper) [inline plugins/in_http/http_prot.c:109]: use a new line..

File: plugins/in_http/http_prot.c [hunk 3]
```diff
@@ -76,37 +76,72 @@ int process_pack(struct flb_http *ctx, flb_sds_t tag, char *buf, size_t size)
...
+            obj = &result.data;
+            for (int i = 0; i < obj->via.array.size; i++)
```
Comment (edsiper) [inline plugins/in_http/http_prot.c:111]: please do not declare a variable on the ```for``` statement, that breaks with old compilers (e.g: centos 7)

**Review by edsiper — CHANGES_REQUESTED** [review]

File: plugins/in_http/http_prot.c (hunk 1 above)
Comment (PettitWesley) [reply plugins/in_http/http_prot.c:87]: oops

**Comment by edsiper** [comment]
once the changes are in place we can merge it.

**Comment by PettitWesley** [comment]
@edsiper Addressed comments. I also added a new commit which adds an option tag_key which allows the user to select the tag based on a key in the record. This is also needed for Lambda support.

**Comment by PettitWesley** [comment]
@edsiper Can we merge this?

**Review by edsiper — APPROVED** [review]

**Pull request merged**